### PR TITLE
feat(client): add `_request` API

### LINKED ISSUE #471: Add more RPC methods to client
Currently the PolkadotClient has a handful of RPC methods - but sadly there isn't the most elemental of them all.. `system.chain()` !! We have `chainSpec_v1_chainName` which is not ubitquitous amongst the rest of Polkadot chains

### PR DESCRIPTION
**Pull request by josepot**
Closes #471

### CODE REVIEW

File: packages/client/src/types.ts [hunk 1]
```diff
@@ -124,6 +124,24 @@ export interface PolkadotClient {
   getTypedApi: <D extends Descriptors>(descriptors: D) => TypedApi<D>
 
   destroy: () => void
+
+  /**
+   * This API is meant as an "ecape hatch" to allow access to debug endpoitns such
+   * as `system_version`, and other useful endpoints that are not spec compliant.
+   *
+   * Example:
```
Comment (carlosala) [inline packages/client/src/types.ts:132]: I'd say `@example`exist👍🏻👍🏻

**Review by carlosala — APPROVED** [review]
Sounds great! 🚀

**Review by voliva — CHANGES_REQUESTED** [review]
I'd like to point out that they could just use substrate-client. If it's for debugging/etc. wouldn't it make sense for them to just use a lower-level subpackage?

Weren't there plans to make the top-level client to take a substrate-client, as an alternative to a JsonRpcProvider?

**Comment by josepot** [comment]
> I'd like to point out that they could just use substrate-client.

`substrate-client` is meant to be used by library authors, its API is too low-level for something like this.

> If it's for debugging/etc. wouldn't it make sense for them to just use a lower-level subpackage?

that's a bit too much IMO.

> Weren't there plans to make the top-level client to take a substrate-client, as an alternative to a JsonRpcProvider? Maybe this is a good chance to do it, in case someone needs to use the both of them.

I've been thinking quite a bit about this and I no longe think that's a good idea. The reason being that it would dramatically increase the API surface for no good reason. In other words: the whole public API of the substrate-client will become part of the public api of the top-level client, making top-level client super leaky.

For instance: as of today, if we need to make a breaking change or a major improvement into `@polkadot-api/substrate-client` we can do so without  affecting in the least the top-level API, which is a pretty good think IMO. That would no longer be the case if we go with that option.

If you think about it, the `_request` is something that allows you to do the most basic kind of interaction possible: a request/reply. Therefore, its API is as simple and straightforward as it can be, which IMO a pretty good compromise for accommodating the most common debugging APIs.

It goes without saying that if someone needs to do something more advanced (like handle notifications/subscriptions), then they will have to do so via a JsonRpcProvider enhancer, which is fine and totally doable, but it's also a bit of a hassle to force everyone to use an enhancer for something this basic.

File: packages/client/src/types.ts [hunk 2]
```diff
@@ -124,6 +124,24 @@ export interface PolkadotClient {
   getTypedApi: <D extends Descriptors>(descriptors: D) => TypedApi<D>
 
   destroy: () => void
+
+  /**
+   * This API is meant as an "escape hatch" to allow access to debug endpoitns such
+   * as `system_version`, and other useful endpoints that are not spec compliant.
+   *
+   * Example:
+   *
```
Comment (josepot) [inline packages/client/src/types.ts:133]:
```suggestion
   * @example
```

**Review by voliva — APPROVED** [review]

**Review by carlosala — APPROVED** [review]

**Pull request merged**